tensorpipe/CMakeLists.txt

```diff
@@ -76,6 +76,19 @@ if(TP_ENABLE_CMA)
   target_compile_definitions(tensorpipe INTERFACE TP_ENABLE_CMA)
 endif()
 
+### cuda
+
+if(TP_ENABLE_CUDA)
+  list(APPEND TENSORPIPE_PUBLIC_HEADERS channel/cuda/context.h)
+  target_sources(tensorpipe PRIVATE
+    channel/cuda/channel.cc
+    channel/cuda/context.cc
+    proto/channel/cuda.proto)
+  target_compile_definitions(tensorpipe INTERFACE TP_ENABLE_CUDA)
+
+  find_package(CUDAToolkit REQUIRED)
```

Comment:
From the CMake docs:
> To find and use the CUDA toolkit libraries the FindCUDAToolkit module has superseded this module [find_package(CUDA)]. It works whether or not the CUDA language is enabled.

Reply:
Just to make sure, from what CMake version does that come? Do you have a link to the docs?

Reply:
From the [CMake 3.17 docs](https://cmake.org/cmake/help/v3.17/module/FindCUDA.html) (which state that `FindCUDA` was deprecated in CMake 3.10.

Reply:
Ok, so PyTorch does that because with CMake 3.5 there's probably nothing else they can do. It may be fine for us to use the modern CMake in our files, but then we'll still have to use the old one in our PyTorch compatibility entry point, and I wonder if that will prevent us from sharing code between the two.

I am wondering whether it may make sense to forgo our own CMake script and just make the PyTorch entry point our main one. We have already seen in the past that we forget to update the PyTorch CMake file and when we do so we happen to break it as it's not covered by CI. It's sad that we won't be able to use modern CMake, but it's not like now we can avoid the ancient one: at the moment we have to use both the modern and the ancient.

Reply:
Updated the PR to be CMake-3.5 compatible.